src/layouts/Pages.jsx

```diff
@@ -21,59 +31,98 @@ export default class Pages extends Component {
+        settingsIsActive: !currState.settingsIsActive,
+        selectedFileName: currState.settingsIsActive ? '' : currState.pages[pageIndex],
+        createNewPage: false,
+      }));
+    }
   }
 
   render() {
-    const { pages, newPageName } = this.state;
-    const { match } = this.props;
+    const {
+      pages, selectedFileName, settingsIsActive, createNewPage,
+    } = this.state;
+    const { match, location } = this.props;
     const { siteName } = match.params;
     return (
-      <div>
-        <Link to="/sites">Back to Sites</Link>
-        <hr />
-        <h2>{siteName}</h2>
-        <ul>
-          <li>
-            <Link to={`/sites/${siteName}/pages`}>Pages</Link>
-          </li>
-          <li>
-            <Link to={`/sites/${siteName}/collections`}>Collections</Link>
-          </li>
-          <li>
-            <Link to={`/sites/${siteName}/images`}>Images</Link>
-          </li>
-          <li>
-            <Link to={`/sites/${siteName}/files`}>Files</Link>
-          </li>
-          <li>
-            <Link to={`/sites/${siteName}/menus`}>Menus</Link>
-          </li>
-        </ul>
-        <hr />
-        <h3>Pages</h3>
-        {pages.length > 0
-          ? pages.map((page) => (
-            <li>
-              <PageCard
-                fileName={page.fileName}
+      <>
+        <Header />
+
+        {/* main bottom section */}
+        <div className={elementStyles.wrapper}>
+          {/* Page settings modal */}
+          { settingsIsActive
+            ? (
+              <PageSettingsModal
+                settingsToggle={this.settingsToggle}
                 siteName={siteName}
+                fileName={selectedFileName}
+                isNewPage={createNewPage}
               />
-            </li>
-          ))
-          : 'No pages'}
-        <br />
-        <input placeholder="New page name" onChange={this.updateNewPageName} />
-        <Link to={`/sites/${siteName}/pages/${newPageName}`}>Create new page</Link>
-      </div>
+            )
+            : null}
+          <Sidebar siteName={siteName} currPath={location.pathname} />
+
+          {/* main section starts here */}
+          <div className={contentStyles.mainSection}>
+            <div className={contentStyles.sectionHeader}>
+              <h1 className={contentStyles.sectionTitle}>Pages</h1>
+              <button
+                type="button"
+                className={elementStyles.blue}
+                id="settings-NEW"
+                onClick={this.settingsToggle}
+              >
+Create New Page
+              </button>
+            </div>
+
+            <div className={contentStyles.contentContainerBars}>
+
+              {/* Page cards */}
+              <ul>
+                <li>
+                  <Link to={`/sites/${siteName}/homepage`}>Homepage</Link>
+                </li>
+                {pages.length > 0
+                  ? pages.map((pageName, pageIndex) => (
+                    <li>
```

Comment:
Should we have a separate component for page and resource cards?

Reply:
Refer to discussion below